File: readH5.m
files = dir('*.h5');
disp('Available H5 files:');
for i = 1:length(files)
    fprintf('%d: %s\n', i, files(i).name);
end

% select a file
disp '======= (^o^) =========';
choice = input('Enter the number of the file you want to plot (such as \"1: 6-AT89S52_ECC.h5\", input 1): ');
filename = files(choice).name;

% read h5 info
info = h5info(filename);

% print details 
stack = {info};
prefixes = {''};

while ~isempty(stack)
    current_info = stack{end};
    current_prefix = prefixes{end};
    stack(end) = [];
    prefixes(end) = [];
    
    if isfield(current_info, 'Groups')
        for i = 1:length(current_info.Groups)
            stack{end+1} = current_info.Groups(i);
            prefixes{end+1} = [current_prefix '/' current_info.Groups(i).Name];
            disp(['Group: ' current_prefix '/' current_info.Groups(i).Name]);
        end
    end
    

    if isfield(current_info, 'Datasets')
        for i = 1:length(current_info.Datasets)
            disp(['Dataset: ' current_prefix '/' current_info.Datasets(i).Name]);
            if contains(current_info.Datasets(i).Name, 'for_paper')
                data = h5read(filename, ['/' current_prefix '/' current_info.Datasets(i).Name]);
                figure;
                plot(data);
                title(strrep(current_info.Datasets(i).Name, '_', ' '));
            end
            
            
        end
    end
end

File: readH5.py
import h5py
import matplotlib.pyplot as plt
import numpy as np
import os

# list all h5 file
h5_files = [f for f in os.listdir('.') if f.endswith('.h5')]
print("Available H5 files:")
for idx, file in enumerate(h5_files, 1):
    print(f"{idx}: {file}")

print("======= (^o^) =========")
# select a file
choice = int(input("Enter the number of the file you want to plot (such as \"1: 6-AT89S52_ECC.h5\", input 1): ")) - 1
file_path = h5_files[choice]

# open h5 file
with h5py.File(file_path, 'r') as file:
    def print_keys(name, obj):
        # print key
        if isinstance(obj, h5py.Dataset):
            print(name)
    def plot_dataset(name, obj):
        if isinstance(obj, h5py.Dataset) and 'for_paper' in name:
            data = np.array(obj)
            plt.figure()
            plt.plot(data)
            plt.title(name)
            plt.show()

    # plot
    file.visititems(print_keys)
    file.visititems(plot_dataset)
